File: ncarg2d/man/gks/gqasf.m
.\"
.\"	$Id: gqasf.m,v 1.16 2008-12-23 00:03:02 haley Exp $
.\"
.TH GQASF 3NCARG "March 1993" UNIX "NCAR GRAPHICS"
.SH NAME
GQASF - (Inquire aspect source flags) - Inquires the values for the 
aspect source flags that determine whether primitive attributes are 
to be selected from individual settings or from "bundle tables".
.SH SYNOPSIS
CALL GQASF (ERRIND,LASF)
.SH C-BINDING SYNOPSIS
#include <ncarg/gks.h>
.sp
void ginq_asfs(Gint *err_ind, Gasfs *list_asf);
.SH DESCRIPTION 
.IP ERRIND 12
(Integer, Output) - If the inquired value cannot be returned correctly,
a non-zero error indicator is returned in ERRIND, otherwise a zero is returned.
Consult "User's Guide for NCAR GKS-0A Graphics" for a description of the
meaning of the error indicators.
.IP LASF 12
(Integer array, output) - An array of settings for the thirteen attributes:
.RS
.IP " 1"
Linetype
.IP " 2"
Linewidth scale factor
.IP " 3"
Polyline color index
.IP " 4"
Marker type
.IP " 5"
Marker size scale factor
.IP " 6"
Polymarker color index
.IP " 7"
Text font and precision
.IP " 8"
Character expansion factor
.IP " 9"
Character spacing
.IP "10"
Text color index
.IP "11"
Fill area interior style
.IP "12"
Fill area style index
.IP "13"
Fill area color index
.RE
.sp
Each value of LASF is either "0" (for "bundled") or "1" (for "individual").
.SH USAGE
In NCAR Graphics all attribute aspect source flags are defaulted to
"individual" (all values for LASF are "1").
It is advised that for dependable results when using NCAR Graphics
these values remain in their default settings.  Some GKS packages
default these values to "bundled".  GQASF can be called to determine
the values of all aspect source flags.
.SH ACCESS
To use GKS routines, load the NCAR GKS-0A library 
ncarg_gks.
.SH SEE ALSO
Online:
gsasf, gsln, gslwsc, gsplci, gsmk, gsmksc, gspmci, gstxfp, gschxp,
gstxsp, gstxci, gsfais, gsfasi, gsfaci, gqln, gqlwsc, gqplci, gqmk, 
gqmksc, gqpmci, gqtxfp, gqchxp, gqtxsp, gqtxci, gqfais, gqfasi, gqfaci,, ginq_asfs
.sp
Hardcopy:  
"User's Guide for NCAR GKS-0A Graphics"
.SH COPYRIGHT
Copyright (C) 1987-2009
.br
University Corporation for Atmospheric Research
.br
The use of this Software is governed by a License Agreement.

File: ncarg2d/src/libncarg_gks/awi/gqasf.f
C
C	$Id: gqasf.f,v 1.5 2008-07-27 00:20:58 haley Exp $
C                                                                      
C                Copyright (C)  2000
C        University Corporation for Atmospheric Research
C                All Rights Reserved
C
C The use of this Software is governed by a License Agreement.
C
      SUBROUTINE GQASF(ERRIND,LASF)
C
C  INQUIRE ASPECT SOURCE FLAGS
C
      include 'gkscom.h'
C
      INTEGER ERRIND,LASF(13)
C
C  Check if GKS is in proper state.
C
      CALL GZCKST(8,-1,ERRIND)
C
      IF (ERRIND.NE.0) THEN
        DO 200 I=1,13
          LASF(I) = -1
  200   CONTINUE
      ELSE
        LASF( 1) = CLNA
        LASF( 2) = CLWSCA
        LASF( 3) = CPLCIA
        LASF( 4) = CMKA
        LASF( 5) = CMKSA
        LASF( 6) = CPMCIA
        LASF( 7) = CTXFPA
        LASF( 8) = CCHXPA
        LASF( 9) = CCHSPA
        LASF(10) = CTXCIA
        LASF(11) = CFAISA
        LASF(12) = CFASIA
        LASF(13) = CFACIA
      ENDIF
C
      RETURN
      END
